pkg/controllers/managementuser/rbac/prtb_handler.go

```diff
@@ -280,63 +280,105 @@ func (m *manager) checkForGlobalResourceRules(role *v3.RoleTemplate, resource st
 	return verbs, nil
 }
 
-// Ensure the clusterRole used to grant access of global resources to users/groups in projects has appropriate rules for the given resource and verbs
-func (m *manager) reconcileRoleForProjectAccessToGlobalResource(resource string, rt *v3.RoleTemplate, newVerbs map[string]bool, baseRule rbacv1.PolicyRule) (string, error) {
-	clusterRoles := m.clusterRoles
-	roleName := rt.Name + "-promoted"
-	if role, err := m.crLister.Get("", roleName); err == nil && role != nil {
-		currentVerbs := map[string]bool{}
-		currentResourceNames := map[string]struct{}{}
-		for _, rule := range role.Rules {
-			if slice.ContainsString(rule.Resources, resource) {
-				for _, v := range rule.Verbs {
-					currentVerbs[v] = true
-				}
-				for _, v := range rule.ResourceNames {
-					currentResourceNames[v] = struct{}{}
-				}
-			}
+// reconcileRoleForProjectAccessToGlobalResource ensure the clusterRole used to grant access of global resources
+// to users/groups in projects has appropriate rules for the given resource and verbs.
+// It returns the created or updated ClusterRole name, or blank "" if none were created or updated.
+// The roleName is used to find and create/update the relevant '<roleName>-promoted' ClusterRole.
+func (m *manager) reconcileRoleForProjectAccessToGlobalResource(resource string, roleName string, newVerbs map[string]struct{}, baseRule rbacv1.PolicyRule) (string, error) {
+	if roleName == "" {
+		return "", errors.New("cannot reconcile Role: missing roleName")
+	}
+	roleName = roleName + "-promoted"
+
+	role, err := m.crLister.Get("", roleName)
+
+	// we try to create the role if not found, or if we get an error (for backward compatibility)
+	if apierrors.IsNotFound(err) || err != nil {
```

Comment:
if the err is not nil and different than not found I think we should return it. Imagine there is a temporary problem getting the role, but the role is there. This code would try to create a new role.

Reply:
This is very right, and it's what I did in the first implementation. But some old tests in a different part were failing, so I kept the old behaviour.
It's in the description of the PR and in the comment above.

Maybe some old folks could confirm if we can change this (@pmatseykanets?)

Reply:
I think it makes sense to return the error when the error is NOT "not found".  If that change winds-up requiring larger changes to the other tests, I suppose it could all be done in a separate PR to keep this one on target.